tools/run_tests/run_xds_tests.py

```diff
@@ -56,17 +56,27 @@
     'secondary_locality_gets_requests_on_primary_failure',
     'traffic_splitting',
 ]
+# Valid test cases, but not in all. So the tests can only run manually, and
+# aren't enabled automatically for all languages. Move them into _TEST_CASES
+# when support is ready in all languages.
+_ALTERNATIVE_TEST_CASES = ['path_matching', 'header_matching']
 
 
 def parse_test_cases(arg):
-    if arg == 'all':
-        return _TEST_CASES
     if arg == '':
         return []
-    test_cases = arg.split(',')
-    if all([test_case in _TEST_CASES for test_case in test_cases]):
-        return test_cases
-    raise Exception('Failed to parse test cases %s' % arg)
+    arg_split = arg.split(',')
+    test_cases = set()
+    all_test_cases = _TEST_CASES + _ALTERNATIVE_TEST_CASES
+    for arg in arg_split:
+        if arg == "all":
+            test_cases = test_cases.union(_TEST_CASES)
+        else:
+            test_cases = test_cases.union([arg])
+    if not all([test_case in all_test_cases for test_case in test_cases]):
```

Comment:
Why change all this versus just keeping this as:

```
if all([test_case in _TEST_CASES or test_case in _ALTERNATIVE_TEST_CASES for test_case in test_cases]):
    return test_cases
raise Exception('Failed to parse test cases %s' % arg)
```

Reply:
Is this comment just to invert `not`?

The union is to avoid having duplicate tests (which will happen when the xds.sh runs `all,path_matching`, and we add `path_matching` into `all`)

We will need `all_test_cases` later to perserve the test orders (not sure if the order will ever matter).

Reply:
This comment was about why this replaces 3 lines of code with ~13. I guess this is because we now support `all` + additional tests, which I overlooked before, so this looks good as-is.